src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -1319,9 +1319,34 @@ void Leader::HandleNetworkDataRestoredAfterReset(void)
 {
     const PrefixTlv *prefix;
     TlvIterator      tlvIterator(GetTlvsStart(), GetTlvsEnd());
+    Iterator         iterator = kIteratorInit;
+    ChangedFlags     flags;
```

Comment:
Should we initialize, in case the network data is empty?

Reply:
The `ChangedFlags` has a constructor which initializes it so I think we should be good.